src/xdp/xsk.c

```diff
@@ -3696,6 +3824,36 @@ XskIrpGetSockopt(
     return Status;
 }
 
+static
+NTSTATUS
+XskIrpGetSockopt(
+    _In_ IRP *Irp,
+    _In_ IO_STACK_LOCATION *IrpSp
+    )
+{
+    NTSTATUS Status = STATUS_SUCCESS;
+    XSK *Xsk;
+    UINT32 Option = 0;
+    UINT32 BufferLength = IrpSp->Parameters.DeviceIoControl.OutputBufferLength;
+
+    Xsk = IrpSp->FileObject->FsContext;
+
+    if (IrpSp->Parameters.DeviceIoControl.InputBufferLength < sizeof(Option)) {
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    Option = *(UINT32 *)Irp->AssociatedIrp.SystemBuffer;
+
+    Status = XskGetSockopt(Xsk, Option, Irp->AssociatedIrp.SystemBuffer, &BufferLength);
+    if (NT_SUCCESS(Status)) {
+        Irp->IoStatus.Information = BufferLength;
+    }
```

Comment:
I've noticed an inconsistency in this patter. Some Irp functions check status before setting the `Information` and some don't (mostly in dispatch.c). Is that on purpose?

Reply:
Yeah I think the pattern is that the IOCTL dispatch initializes Information to 0 (dispatch.c), and the individual IOCTL handlers set Information only if the IOCTL succeeds.